File: NATSInternal.Web/Models/Shared/StockBasicModel.cs
using System.ComponentModel;
using NATSInternal.Application.Localization;
using NATSInternal.Application.UseCases.Shared;

namespace NATSInternal.Web.Models;

public class StockBasicModel
{
    #region Constructors
    public StockBasicModel(StockBasicResponseDto responseDto)
    {
        Id = responseDto.Id;
        StockingQuantity = responseDto.StockingQuantity;
        ResupplyThresholdQuantity = responseDto.ResupplyThresholdQuantity;
    }
    #endregion

    #region Properties
    public Guid Id { get; }
    
    [DisplayName(DisplayNames.StockingQuantity)]
    public int StockingQuantity { get; }
    
    [DisplayName(DisplayNames.ResupplyThresholdQuantity)]
    public int? ResupplyThresholdQuantity { get; }
    #endregion
}

File: NATSInternal.Frontend/NATSInternal.Vue/src/models/shared/stockBasicModel.ts
declare global {
  type StockBasicModel = {
    id: string;
    stockingQuantity: number;
    resupplyThresholdQuantity: number;
  };
}

export function createStockBasicModel(responseDto: StockBasicResponseDto): StockBasicModel {
  return {
    id: responseDto.id,
    stockingQuantity: responseDto.stockingQuantity,
    resupplyThresholdQuantity: responseDto.resupplyThresholdQuantity
  };
}